backend/service/chaos/experimentation/experimentstore/storer.go

```diff
@@ -260,29 +241,56 @@ func (s *storer) GetListView(ctx context.Context) ([]*experimentation.ListViewIt
 
 func (s *storer) GetExperimentRunDetails(ctx context.Context, id uint64) (*experimentation.ExperimentRunDetails, error) {
 	sqlQuery := `
-        SELECT experiment_run.id, lower(execution_time), upper(execution_time), cancellation_time, creation_time, details FROM experiment_config, experiment_run
+		SELECT 
+			experiment_run.id,
+			lower(execution_time),
+			upper(execution_time),
+			cancellation_time,
+			creation_time,
+			experiment_config.id, 
+			details FROM experiment_config, experiment_run
         WHERE experiment_run.id = $1 AND experiment_run.experiment_config_id = experiment_config.id`
 
 	row := s.db.QueryRowContext(ctx, sqlQuery, id)
 
-	var fetchedID uint64
-	var startTime, endTime, cancellationTime sql.NullTime
-	var creationTime time.Time
 	var details string
+	run := ExperimentRun{}
+	config := ExperimentConfig{Config: &any.Any{}}
+	err := row.Scan(&run.id, &run.startTime, &run.endTime, &run.cancellationTime, &run.creationTime, &config.id, &details)
+	if err != nil {
+		return nil, err
+	}
 
-	err := row.Scan(&fetchedID, &startTime, &endTime, &cancellationTime, &creationTime, &details)
+	err = jsonpb.Unmarshal(strings.NewReader(details), config.Config)
 	if err != nil {
 		return nil, err
 	}
 
-	return NewRunDetails(fetchedID, creationTime, startTime, endTime, cancellationTime, time.Now(), details)
+	return NewRunDetails(&run, &config, s.transformer, time.Now())
 }
 
 // Close closes all resources held.
 func (s *storer) Close() {
 	s.db.Close()
 }
 
+func (s *storer) RegisterTransformation(typeUrl string, transformation func(*ExperimentConfig) ([]*experimentation.Property, error)) {
+	s.transformer.nameToConfigTransformMap[typeUrl] = transformation
+}
+
+func toProto(t *time.Time) (*timestamp.Timestamp, error) {
+	if t == nil {
+		return nil, nil
+	}
```

Comment:
If the caller is expecting a `timestamp` why not throw an error here?

Reply:
I just copied this method over from https://github.com/lyft/clutch/pull/499/files#diff-c4a96668461e1e041b98305db571af49L127 (a few lines above). This method handles the nil time case so that the code on the caller side can be simpler I guess. A nil time is a valid input. What would be your recommendation here (I can implement it as a part of a separate PR)?

Reply:
The caller should check nil before invoking this function, if they dont need a proto for it. Feels weird that we would would `return nil, nil`. You can have a follow up PR to change this.